File: spl-net/src/main/java/bgu/spl171/net/impl/TFTP/CommandBuilders/AckCommandBuilder.java
package bgu.spl171.net.impl.TFTP.CommandBuilders;

import bgu.spl171.net.impl.TFTP.Commands.Command;
import bgu.spl171.net.impl.TFTP.Commands.Responses.Ack;
import bgu.spl171.net.impl.TFTP.EncoderDecoders.ShortEncoderDecoder;

public class AckCommandBuilder implements CommandBuilder {
    ShortEncoderDecoder shortEncoderDecoder = new ShortEncoderDecoder();

    @Override
    public Command decodeNextByte(byte nextByte) {
        Short blockId = shortEncoderDecoder.decodeNextByte(nextByte);
        if(blockId!=null){
            return new Ack(blockId);
        }
        return null;
    }
}

File: java to c++ builders/AckCommandBuilder.cpp
#include "AckCommandBuilder.h"

namespace bgu
{
	namespace spl171
	{
		namespace net
		{
			namespace impl
			{
				namespace TFTP
				{
					namespace CommandBuilders
					{
						using bgu::spl171::net::impl::TFTP::Commands::Command;
						using bgu::spl171::net::impl::TFTP::Commands::Responses::Ack;
						using bgu::spl171::net::impl::TFTP::EncoderDecoders::ShortEncoderDecoder;

						Command *AckCommandBuilder::decodeNextByte(char nextByte)
						{
							boost::optional<short> blockId = shortEncoderDecoder->decodeNextByte(nextByte);
							if (blockId != nullptr)
							{
								return new Ack(blockId);
							}
							return nullptr;
						}
					}
				}
			}
		}
	}
}
